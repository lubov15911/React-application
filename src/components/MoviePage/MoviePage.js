import React from 'react';
import PropTypes from 'prop-types';

import HeaderContainer from '../HeaderContainer';
import SearchResultsContainer from '../SearchResultsContainer';

export default class MoviePage extends React.PureComponent {
    render() {
        const {
            films,
            goToMoviePage,
        } = this.props;

        return (
            <>
                <HeaderContainer movieData={films[5]} />
                <SearchResultsContainer
                    total={films.length}
                    films={films}
                    raiseClickEvent={goToMoviePage} />
            </>
        );
    }
};
MoviePage.propTypes = {
    films: PropTypes.arrayOf(PropTypes.object).isRequired,
    goToMoviePage: PropTypes.func.isRequired,
};
