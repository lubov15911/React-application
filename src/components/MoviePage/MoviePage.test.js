import React from 'react';
import { shallow } from 'enzyme';

import MoviePage from './MoviePage';

describe('MoviePage', () => {
    const movieId = 354912;
    const Films = { data: [] };
    const fakeFunction = () => {};

    let component;

    beforeEach(() => {
        component = shallow(<MoviePage
            movieId={movieId}
            films={Films.data}
            routeToHomePage={fakeFunction}
            goToMoviePage={fakeFunction}/>);
    });

    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should change sortOption state', () => {
        component.find('SearchResultsContainer').props().handleSortOption();
        expect(component.state().sortOption).toBe('Rating');
        component.find('SearchResultsContainer').props().handleSortOption();
        expect(component.state().sortOption).toBe('Release date');
    });
});
