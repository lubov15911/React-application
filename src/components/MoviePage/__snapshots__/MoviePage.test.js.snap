// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should render correctly 1`] = `
<Fragment>
  <HeaderContainer
    handleSearchValue={null}
    movieData={null}
    searchOption={null}
    toggleSearch={null}
  />
  <SearchResultsContainer
    films={Array []}
    raiseClickEvent={[Function]}
    total={0}
  />
</Fragment>
`;
